docs/log_rotation.md

```diff
@@ -0,0 +1,34 @@
+# Log Rotation
+
+This doc outlines how to rotate the logs detailed in [transparency.md](transparency.md).
+
+## Steps
+
+1. [Rotate](https://cloud.google.com/kms/docs/rotate-key#manual) the log key to create a new key version. Note: this can be auotmated in Terraform, but we do not currently manage the key resources in Terraform.
+
+1. Increment the `bucket_count` to create new firmware artefact and log buckets in GCS, `log_shard` to start writing to the new buckets, and `log_public_key` to match the new keys.
+    * Example [PR](https://github.com/transparency-dev/armored-witness/pull/182/)
+    * The `log_public_key` needs to be generated from the [public key](https://cloud.google.com/kms/docs/retrieve-public-key) of the new key version.
+    * After applying the [`build_and_release`](/deployment/build_and_release) Terraform changes, the output values will be used for new paths at https://api.transparency.dev (step #4).
+
+1. Initialize the log bucket:
+    ```
+    gcloud functions call integrate \
+    --data '{
+        "initialise": true,
+        "origin": "transparency.dev/armored-witness/firmware_transparency/$ENV/$LOG_SHARD",
+        "bucket": "armored-witness-firmware-log-$ENV-$LOG_SHARD",
+        "kmsKeyName": "ft-log-$ENV",
+        "kmsKeyRing": "firmware-release-$ENV",
+        "kmsKeyVersion": $LOG_SHARD,
+        "kmsKeyLocation": "global",
+        "noteKeyName": "transparency.dev-aw-ftlog-$ENV-$LOG_SHARD"
+    }'
+    ```
+1. Apply [`api_transparency_dev`](/deployment/api_transparency_dev) Terraform to create new https://api.transparency.dev paths for the firmware artefacts and log.
+1. Populate the buckets by running the build triggers.
```

Comment:
We should probably have a step in here to ensure that all 4 types of firmware are present in the new log before continuing on to update the tooling deps below.

Could be a manual "go click Run on the 4 triggers before continuing", but I imagine we can probably automate that too somehow?

Reply:
Aha, great!

One suggestion, maybe change the list item to make it clearer about what we're doing here and why, e.g. "Populate the new log with all four firmware types so new devices can straightaway be provisioned from it." or something?

Reply:
Done.